### PR TITLE
Update pod pages to v1beta3 in web ui

### PR DESCRIPTION
**Pull request by bcbroussard**
Depends on https://github.com/GoogleCloudPlatform/kubernetes/pull/8304

cc @jackgr @lavalamp @bgrant0607 @brendandburns

### CODE REVIEW

File: www/app/assets/js/app.js [hunk 1]
```diff
@@ -1313,34 +1362,34 @@ app.controller('ListPodsCtrl', [
...
 
           $scope.content.push({
-            pod: pod.id,
-            ip: pod.currentState.podIP,
+            pod: pod.metadata.name,
+            ip: pod.status.podIP,
             containers: _fixComma(_containers),
             images: _fixComma(_images),
-            host: pod.currentState.host,
+            host: pod.status.hostIP,
```
Comment (bgrant0607) [inline www/app/assets/js/app.js]: Why not pod.spec.host? That's the field set by the scheduler.

File: www/app/assets/js/app.js [hunk 2]
```diff
@@ -2195,20 +2244,20 @@ angular.module('kubernetesApp.components.dashboard')
     var pods = {
       "kind": "PodList",
       "creationTimestamp": null,
-      "selfLink": "/api/v1beta1/pods",
+      "selfLink": "/api/v1beta3/pods",
```
Comment (bgrant0607) [inline www/app/assets/js/app.js]: This is all still v1beta1 -- desiredState, manifest, ... The v1beta3 schema is different. The example needs to be regenerated.

File: www/app/assets/js/app.js [hunk 3]
```diff
@@ -2274,9 +2323,9 @@ angular.module('kubernetesApp.components.dashboard')
     var replicationControllers = {
       "kind": "ReplicationControllerList",
       "creationTimestamp": null,
-      "selfLink": "/api/v1beta1/replicationControllers",
+      "selfLink": "/api/v1beta3/replicationcontrollers",
```
Comment (bgrant0607) [inline www/app/assets/js/app.js]: This is still using the v1beta1 schema.

File: www/app/assets/js/app.js [hunk 4]
```diff
@@ -2308,15 +2357,15 @@ angular.module('kubernetesApp.components.dashboard')
     var services = {
       "kind": "ServiceList",
       "creationTimestamp": null,
-      "selfLink": "/api/v1beta1/services",
+      "selfLink": "/api/v1beta3/services",
```
Comment (bgrant0607) [inline www/app/assets/js/app.js]: This is also still using the v1beta1 schema.

File: www/app/components/dashboard/views/pod.html [hunk 5]
```diff
@@ -4,54 +4,48 @@
...
 
 
-      <table>
+      <table class="align-top">
         <tbody>
           <tr>
             <td class="name">Status</td>
             <td class="value">
-              {{pod.currentState.status}} on <a ng-href="#/dashboard/groups/host/selector/host={{pod.currentState.host}}">{{pod.currentState.host}}</a>
+              {{pod.status.phase}} on <a ng-href="#/dashboard/groups/host/selector/host={{pod.status.hostIP}}">{{pod.status.hostIP}}</a>
```
Comment (bgrant0607) [inline www/app/components/dashboard/views/pod.html]: pod.spec.host

File: www/master/components/dashboard/js/modules/controllers/listPodsController.js [hunk 6]
```diff
@@ -71,34 +71,34 @@ app.controller('ListPodsCtrl', [
...
 
           $scope.content.push({
-            pod: pod.id,
-            ip: pod.currentState.podIP,
+            pod: pod.metadata.name,
+            ip: pod.status.podIP,
             containers: _fixComma(_containers),
             images: _fixComma(_images),
-            host: pod.currentState.host,
+            host: pod.status.hostIP,
```
Comment (bgrant0607) [inline www/master/components/dashboard/js/modules/controllers/listPodsController.js]: pod.spec.host

File: www/master/components/dashboard/js/modules/services/podsMock.js [hunk 7]
```diff
@@ -14,20 +14,20 @@
...
+      "selfLink": "/api/v1beta3/pods",
       "resourceVersion": 166552,
-      "apiVersion": "v1beta1",
+      "apiVersion": "v1beta3",
       "items": [{
         "id": "hello",
         "uid": "0fe3644e-ab53-11e4-8ae8-061695c59fcf",
         "creationTimestamp": "2015-02-03T03:16:36Z",
-        "selfLink": "/api/v1beta1/pods/hello?namespace=default",
+        "selfLink": "/api/v1beta1/pods/hello",
```
Comment (bgrant0607) [inline www/master/components/dashboard/js/modules/services/podsMock.js]: v1beta1 still

File: www/master/components/dashboard/js/modules/services/podsMock.js [hunk 8]
```diff
@@ -40,7 +40,7 @@
             "dnsPolicy": "ClusterFirst"
           }
         },
-        "currentState": {
+        "status": {
```
Comment (bgrant0607) [inline www/master/components/dashboard/js/modules/services/podsMock.js]: v1beta1 still

File: www/master/components/dashboard/js/modules/services/servicesMock.js [hunk 9]
```diff
@@ -14,15 +14,15 @@
     var services = {
       "kind": "ServiceList",
       "creationTimestamp": null,
-      "selfLink": "/api/v1beta1/services",
+      "selfLink": "/api/v1beta3/services",
       "resourceVersion": 166552,
-      "apiVersion": "v1beta1",
+      "apiVersion": "v1beta3",
```
Comment (bgrant0607) [inline www/master/components/dashboard/js/modules/services/servicesMock.js]: v1beta1 schema

File: www/master/components/dashboard/js/modules/services/servicesMock.js [hunk 10]
```diff
@@ -14,15 +14,15 @@
...
       "items": [
         {
           "id": "kubernetes",
           "uid": "626dd08d-ab51-11e4-8ae8-061695c59fcf",
           "creationTimestamp": "2015-02-03T03:04:36Z",
-          "selfLink": "/api/v1beta1/services/kubernetes?namespace=default",
+          "selfLink": "/api/v1beta3/services/kubernetes",
```
Comment (bgrant0607) [inline www/master/components/dashboard/js/modules/services/servicesMock.js]: v1beta1 schema

Also, namespace must be in the path in v1beta3.

File: www/master/components/dashboard/views/pod.html [hunk 11]
```diff
@@ -4,54 +4,48 @@
...
 
 
-      <table>
+      <table class="align-top">
         <tbody>
           <tr>
             <td class="name">Status</td>
             <td class="value">
-              {{pod.currentState.status}} on <a ng-href="#/dashboard/groups/host/selector/host={{pod.currentState.host}}">{{pod.currentState.host}}</a>
+              {{pod.status.phase}} on <a ng-href="#/dashboard/groups/host/selector/host={{pod.status.hostIP}}">{{pod.status.hostIP}}</a>
```
Comment (bgrant0607) [inline www/master/components/dashboard/views/pod.html]: pod.spec.host

**Comment by bcbroussard** [comment]
@bgrant0607 - The pod files have been updated to use host instead of hostIP, and the mocks have been changed in the other PR #8304 as well.

File: www/app/assets/js/app.js [hunk 12]
```diff
@@ -1350,8 +1399,8 @@ app.controller('ListPodsCtrl', [
 
     $scope.getPodRestarts = function(pod) {
       var r = null;
-      var container = _.first(pod.desiredState.manifest.containers);
-      if (container) r = pod.currentState.info[container.name].restartCount;
+      var container = _.first(pod.spec.containers);
+      if (container) r = pod.status.info[container.name].restartCount;
```
Comment (caesarxuchao) [inline www/app/assets/js/app.js]: pod.status.ContainerStatuses[container.name].restartCount

**Comment by caesarxuchao** [comment]
@bcbroussard, I think I have review all your PRs that are assigned to me. Ping me if I left any.

**Comment by bcbroussard** [comment]
Thanks @caesarxuchao. I've made those two updates.

**Comment by caesarxuchao** [comment]
Thank you @bcbroussard, LGTM

**Comment by thockin** [comment]
needs rebase

**Comment by bgrant0607** [comment]
I'd like to get all of the v1beta3-related PRs ASAP, prior to the 0.18 cut. Please rebase.

**Comment by thockin** [comment]
rebase

**Comment by bcbroussard** [comment]
@thockin rebased

**Pull request merged**